### PR TITLE
Propose to serve boulderwiki.org directly

### PR DESCRIPTION
**Pull request by robacarp**
Please double check for me before/after merging:
- paths to ssl_cert, key, and trusted cert (wild guess on all three)
- line 88, I'm not 100% sure that I don't want a slash between ```boulderwiki.org``` and ```$request_uri```

### CODE REVIEW

File: modules/mediawiki/files/nginx/mediawiki.conf [hunk 1]
```diff
@@ -22,7 +22,7 @@ server {
 server {
 	listen [::]:80;
 
-	server_name *.miraheze.org anuwiki.com spiral.wiki static.miraheze.org antiguabarbudacalypso.com permanentfuturelab.wiki secure.reviwiki.info wiki.printmaking.be private.revi.wiki oneagencydunedin.wiki publictestwiki.com;
+	server_name *.miraheze.org anuwiki.com spiral.wiki static.miraheze.org antiguabarbudacalypso.com permanentfuturelab.wiki secure.reviwiki.info wiki.printmaking.be private.revi.wiki oneagencydunedin.wiki publictestwiki.comboulderwiki.com www.boulderwiki.com;
```
Comment (Southparkfan) [inline modules/mediawiki/files/nginx/mediawiki.conf]: publictestwiki.com and boulderwiki.org should be separated by a space.

File: modules/mediawiki/files/nginx/mediawiki.conf [hunk 2]
```diff
@@ -63,6 +63,30 @@ server {
...
+	add_header Strict-Transport-Security "max-age=2419200";
+
+	include /etc/nginx/mediawiki-includes;
+}
+
+server {
+	# Redirect www.boulderwiki to straight boulderwiki
+	listen [::]:443 ssl spdy;
+	server_name www.boulderwiki.org;
+	return 301 https://boulderwiki.org$request_uri;
```
Comment (Southparkfan) [inline modules/mediawiki/files/nginx/mediawiki.conf]: You might want to look at lines 170-172.

Comment (robacarp) [inline modules/mediawiki/files/nginx/mediawiki.conf]: @Southparkfan I'm not sure what you mean. Are you talking about this location directive?

```
 	location ~ ^(.*)$ {
 		return 301 https://private.revi.wiki$1;
 	}
```

It seems unnecessary to force a regex match to all urls if it's just going to do the same thing for all urls. In my testing elsewhere, this redirect works as expected.

File: modules/varnish/files/nginx/mediawiki.conf [hunk 3]
```diff
@@ -47,6 +47,31 @@ server {
...
+                proxy_pass http://127.0.0.1:81;
+                proxy_set_header Host $host;
+                proxy_set_header X-Real-IP $remote_addr;
+                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+                proxy_set_header X-Forwarded-Proto $scheme;
+        }
+}
+
+server {
+        listen [::]:80 ipv6only=off;
```
Comment (Southparkfan) [inline modules/varnish/files/nginx/mediawiki.conf]: Please remove the unnecessary "ipv6only=off" here.

File: modules/varnish/files/nginx/mediawiki.conf [hunk 4]
```diff
@@ -47,6 +47,31 @@ server {
         }
 }
 
+server {
+        listen [::]:443 ssl spdy ipv6only=off;
```
Comment (Southparkfan) [inline modules/varnish/files/nginx/mediawiki.conf]: Please remove the unnecessary "ipv6only=off" here.

File: modules/mediawiki/files/nginx/mediawiki.conf [hunk 5]
```diff
@@ -22,7 +22,7 @@ server {
 server {
 	listen [::]:80;
 
-	server_name *.miraheze.org anuwiki.com spiral.wiki static.miraheze.org antiguabarbudacalypso.com permanentfuturelab.wiki secure.reviwiki.info wiki.printmaking.be private.revi.wiki oneagencydunedin.wiki publictestwiki.com;
+	server_name *.miraheze.org anuwiki.com spiral.wiki static.miraheze.org antiguabarbudacalypso.com permanentfuturelab.wiki secure.reviwiki.info wiki.printmaking.be private.revi.wiki oneagencydunedin.wiki publictestwiki.com boulderwiki.com www.boulderwiki.com;
```
Comment (Southparkfan) [inline modules/mediawiki/files/nginx/mediawiki.conf]: boulderwiki.com?

**Pull request merged**